src/app/app.component.html

```diff
@@ -0,0 +1,55 @@
+<div class="container">
+  <form method="post" class="form" [formGroup]="form" (ngSubmit)="onSubmit()">
+    <mat-form-field appearance="fill">
+      <mat-label>Name</mat-label>
+      <input matInput required formControlName="firstName" />
+    </mat-form-field>
+
+    <mat-form-field appearance="fill">
+      <mat-label>Last name</mat-label>
+      <input matInput required formControlName="lastName" />
+    </mat-form-field>
+
+    <mat-form-field appearance="fill">
+      <mat-label>Choose a date</mat-label>
+      <input matInput [matDatepicker]="picker" formControlName="dateOfBirth" />
+      <mat-hint>MM/DD/YYYY</mat-hint>
+      <mat-datepicker-toggle matSuffix [for]="picker"></mat-datepicker-toggle>
+      <mat-datepicker #picker></mat-datepicker>
+    </mat-form-field>
+
+    <mat-form-field appearance="fill">
+      <mat-label>FE technology</mat-label>
+      <mat-select formControlName="framework" (valueChange)="onChange($event)">
```

Comment:
in angular there is more common way to subscribe on formControl changes -> this.form.get(control).valueChanges
can be done in ngOnInit
let's implement it instead of valueChange event